### DISCUSSION TITLE
Separate training from inference more cleanly in Spaceflights?

### DISCUSSION THREAD

**Discussion by astrojuanlu** [kedro-org/kedro#3922] [Idea]
Today I wanted to do a inference-only run with a Spaceflights project (see #3856) and I had to resort to this:

1. Declare `X_test`, `y_test` in the catalog (and found a scikit-learn bug along the way https://github.com/kedro-org/kedro/issues/3674#issuecomment-2147185329)
2. Training: `kedro run --to-outputs=X_test,y_test,regressor`
3. Actual inference: `kedro run --from-nodes=evaluate_model_node`

It would be good to rework our Spaceflights starters so that they follow an architecture that more closely resembles [Hopsworks' FTI Pipeline model](https://www.hopsworks.ai/post/mlops-to-ml-systems-with-fti-pipelines), which would in turn make it easier for users (and for ourselves) to see how could they

- Easily transition to a production-like setting
- Perform inference with another set of data that was not the training or evaluation data
- Serve the model they trained, with Kedro or something else (MLflow, Seldon Core, a custom FastAPI endpoint)

![image](https://github.com/kedro-org/kedro/assets/316517/5e184b8f-e4e3-4e94-b702-cb6d5b214bb7)

**Comment by DimedS** [comment]
Nice article, I agree with that
